agent_app_sample_code/05_evaluate_poc_quality.py

```diff
@@ -0,0 +1,397 @@
+    (eval_results_df["retrieval/llm_judged/chunk_relevance/precision"] < 0.5)
+    & (eval_results_df["response/llm_judged/groundedness/rating"] == "no")
+    & (eval_results_df["response/llm_judged/relevance_to_query/rating"] == "yes"),
+    (eval_results_df["retrieval/llm_judged/chunk_relevance/precision"] < 0.5)
+    & (eval_results_df["response/llm_judged/groundedness/rating"] == "yes")
+    & (eval_results_df["response/llm_judged/relevance_to_query/rating"] == "no"),
+    (eval_results_df["retrieval/llm_judged/chunk_relevance/precision"] < 0.5)
+    & (eval_results_df["response/llm_judged/groundedness/rating"] == "yes")
+    & (eval_results_df["response/llm_judged/relevance_to_query/rating"] == "yes"),
+    (eval_results_df["retrieval/llm_judged/chunk_relevance/precision"] >= 0.5)
+    & (eval_results_df["response/llm_judged/groundedness/rating"] == "no")
+    & (eval_results_df["response/llm_judged/relevance_to_query/rating"] == "no"),
+    (eval_results_df["retrieval/llm_judged/chunk_relevance/precision"] >= 0.5)
+    & (eval_results_df["response/llm_judged/groundedness/rating"] == "no")
+    & (eval_results_df["response/llm_judged/relevance_to_query/rating"] == "yes"),
+    (eval_results_df["retrieval/llm_judged/chunk_relevance/precision"] >= 0.5)
+    & (eval_results_df["response/llm_judged/groundedness/rating"] == "yes")
+    & (eval_results_df["response/llm_judged/relevance_to_query/rating"] == "no"),
+    (eval_results_df["retrieval/llm_judged/chunk_relevance/precision"] >= 0.5)
+    & (eval_results_df["response/llm_judged/groundedness/rating"] == "yes")
+    & (eval_results_df["response/llm_judged/relevance_to_query/rating"] == "yes"),
+]
+
+root_causes = [
+    "Improve Retrieval",
+    "Improve Retrieval",
+    "Improve Retrieval",
+    "Improve Retrieval",
+    "Improve Generation",
+    "Improve Generation",
+    "Improve Generation",
+    "No root cause, record passed",
+]
+
+overall_ratings = ["fail", "fail", "fail", "pass", "fail", "fail", "fail", "pass"]
+
+root_cause_rationales = [
+    "Retrieval quality is poor.",
+    "Retrieval quality is poor.",
+    "Response is grounded in retrieval, but retrieval is poor.",
+    "Relevant response grounded in the retrieved context and relevant, but retrieval is poor.",
+    "Hallucination",
+    "Hallucination",
+    "Good retrieval & grounded, but LLM does not provide a relevant response.",
+    "Good retrieval and relevant response. Collect ground-truth to know if the answer is correct.",
+]
+
+
+# Create new columns in the dataframe based on the conditions
+eval_results_df["overall/root_cause/rating"] = pd.Series(
+    pd.NA, index=eval_results_df.index
+)
+eval_results_df["overall/assessment"] = pd.Series(pd.NA, index=eval_results_df.index)
+eval_results_df["overall/root_cause/rationale"] = pd.Series(
+    pd.NA, index=eval_results_df.index
+)
+
+for condition, root_cause, overall_rating, root_cause_rationale in zip(
```

Comment:
These look like they should be pushed into the SDK or similar, vs in user code - seems quite painful to have to define these yourself as a user, maybe we can follow up with folks working on agent eval

Reply:
afaik this is downstream of the agents SDK. at this point user has results stored in MLflow as a table and is just looking at a raw pandas object. 

i do think we should improve this CUJ in the longer term, but i think this is probably out of socpe for now.